client/src/main/java/io/pravega/client/tables/impl/TableSegment.java

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.tables.impl;
+
+import com.google.common.annotations.Beta;
+import io.pravega.client.tables.ConditionalTableUpdateException;
+import io.pravega.client.tables.KeyVersion;
+import io.pravega.client.tables.TableEntry;
+import io.pravega.client.tables.TableKey;
+import io.pravega.common.util.AsyncIterator;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import lombok.Data;
+
+/**
+ * Defines all operations that are supported on a Table Segment.
```

Comment:
Where do we explain what a table segment is? The concept needs to be defined somewhere, and I'd expect it to be here.

Reply:
This was explained in the main `Table` API, which has been removed as part of this PR.